src/udev/udev-varlink.c

```diff
@@ -73,6 +73,35 @@ static int vl_method_start_exec_queue(Varlink *link, JsonVariant *parameters, Va
         return update_exec_queue(link, parameters, userdata, /* stop = */ false);
 }
 
+static int vl_method_set_environment(Varlink *link, JsonVariant *parameters, VarlinkMethodFlags flags, void *userdata) {
+        static const JsonDispatch dispatch_table[] = {
+                {"assignment", JSON_VARIANT_STRING, json_dispatch_const_string, 0, JSON_MANDATORY},
```

Comment:
hmm, if you touch this, then please clean this up. i.e. split this in two calls "setEnvironment" (which takes two strings, key and value) and "unsetEnvironment" (which takes one param, just the key).

the old protocol couldn't distinguis between "unset var" and "set var to empty string", I think we should normalize this in the protocol if we modernize it anyway.

Reply:
Yeah, I've thought about doing this. Even asked about it in the initial comment...

Another API-related question: Should `StartExecQueue()` and `StopExecQueue()` be merged into a single method with a bool parameter?

Reply:
i'd keep them separate